### ISSUE TITLE
user defined sampler

### ISSUE THREAD

**Issue by pengzhangzhi** [NUS-HPC-AI-Lab/InfoBatch#15]
Hi, the InfoBatch wraps the dataset and generates a sampler for dataloader. What if I have a pre-defined sampler? How to mitigate my sampler to the InfoBatch's sampler?

**Comment by tiandunx** [comment]
The essence of this method is to achieve lossless acceleration by randomly discarding well-learned samples and rescaling the gradients of the discarded samples. Therefore, you can customize the implementation based on the ideas presented in the paper.

**Comment by henryqin1997** [comment]
PS: you can refer to the research/ directory for a more straightforward implementation of the code. If you are using distributed training, there are some more points to take care of. If you have further questions, you can post them in detail.
